src/nr_openai_observability/bedrock.py

```diff
@@ -0,0 +1,412 @@
+import botocore
+import json
+import logging
+import newrelic.agent
+import sys
+import time
+import uuid
+
+from anthropic import _tokenizers
+from datetime import datetime
+from nr_openai_observability.monitor import monitor
+from nr_openai_observability.patcher import _patched_call
+from nr_openai_observability.consts import EventName, MessageEventName, SummaryEventName, TransactionBeginEventName
+from nr_openai_observability.error_handling_decorator import handle_errors
+
+
+logger = logging.getLogger("nr_openai_observability")
+logger.addHandler(logging.StreamHandler(sys.stdout))
+logger.setLevel(logging.INFO)
+
+def perform_patch_bedrock():
+    from newrelic.common.package_version_utils import get_package_version_tuple
+
+    (major, minor, revision) = get_package_version_tuple('botocore')
+
+    if major != 1 and minor < 31 and revision < 57:
```

Comment:
Does this handle cases like `1.30.58`? Maybe we need something like `major < 1 or minor < 31 or (minor == 31 and revision < 57)`?

Reply:
Nice catch. There has to be a python package that handles version checks for us, right? :)

Reply:
I am applying your suggested change for now. We'll want to find a better way that allows for clearer version comparisons. There are a number of changes (such as in setuptools) that already run these checks. It'll take a bit to get a good working solution.